src/redata.js

```diff
@@ -1,15 +1,41 @@
+/**
+ * redata's data object definition
+ * @typedef {Object} Data
+ * @property {boolean} loading - true if the loader is still running, false otherwise.
+ * @property {error} error - instance of Error in case the loader failed, undefined otherwise.
+ * @property {*} result - result of the loader, or undefined if the loader is still running.
+ */
+
+/**
+ * Loader function argument structure
+ * @typedef {Object} LoaderArguments
```

Comment:
I think we should avoid specifying the params structure, since redata is quite generic on this level. props, nextProps, etc, are quite specific for the React implementation.